opts/opts.go

```diff
@@ -263,6 +263,16 @@ func ValidateLabel(val string) (string, error) {
 	return val, nil
 }
 
+// ValidateSingleGenericResource validates that a single entry in the
+// generic resource list is valid.
+// i.e 'GPU=UID1' is valid however 'GPU:UID1' or 'UID1' isn't
+func ValidateSingleGenericResource(val string) (string, error) {
+	if strings.Count(val, "=") < 1 {
+		return "", fmt.Errorf("bad attribute format `%s` expected argument of the for `name=value`", val)
```

Comment:
This message looks incomplete ("argument of the for .. ")?

Looking at other messages, I think we use `invalid ...` for other similar messages (e.g. https://github.com/moby/moby/blob/d91c5f42eb37c6f88cec4021c10c0a1ded1785c3/daemon/cluster/convert/container.go#L296)

Perhaps something like:

```
invalid node-generic-resource format: expected 'name=value':
```

@mistyhacks perhaps you have a better suggestion?

Reply:
I think it's ok, if you also include a `got` portion to the error message.  What is a "node-generic-resource"?

Reply:
@mistyhacks the full design document [is here](https://github.com/docker/swarmkit/blob/master/design/generic_resources.md).
The basic idea is that it's a way for a node to advertise user-defined resources